libraries/libfc/src/string.cpp

```diff
@@ -83,13 +85,42 @@ namespace fc  {
   std::string trim( const std::string& s )
   {
       return boost::algorithm::trim_copy(s);
-      /*
-      std::string cpy(s);
-      boost::algorithm::trim(cpy);
-      return cpy;
-      */
   }
 
+  std::pair<std::string&, bool> escape_str( std::string& str, bool escape_control_chars,
+                                            std::size_t max_len, const char* add_truncate_str )
```

Comment:
A suggestion only:

```suggestion
  enum class escape_control_chars { off, on };
  std::pair<std::string&, bool> escape_str( std::string& str, escape_control_chars escape_ctrl,
                                            std::size_t max_len, const char* add_truncate_str )
```

Which allows you to call it as:

`fc::escape_str(details, fc::escape_control_chars::on, 1024);`

which is nicer than having to declare a bool for clarity.

Reply:
Done.